src/segger/data/io.py

```diff
@@ -505,6 +545,12 @@ def check_overlap(transcript, polygons_gdf):
         # Apply the check_overlap function in parallel to each row using Dask's map_partitions
         if self.verbose:
             print(f"Starting overlap computation for transcripts with the boundary polygons.")
+        if isinstance(transcripts_df, pd.DataFrame):
```

Comment:
I think I found a bug here. Because in the parent function `transcripts_df` becomes (sometimes?) a `pd.DataFrame`, while the code below wants a Dask dataframe. I quickly hacked this non-solution that just makes the code run. It's not a solution because, as explained in the comment, it doesn't lead to a performance improvement by switching to a Dask dataframe.